File: Praktikum 4. Loop/Soal 3/PRAK403-2310817310001-AvantioFierzaPatria.py
angka1, angka2 = map(int, input(
    "Masukkan dua angka yang dipisahkan dengan spasi: ").split())

batasan_awal, batasan_akhir = angka1, angka2

if angka1 > angka2:
    i = angka1 - angka2
else:
    i = angka2 - angka1

awal, akhir = 1, i + 1
print("Angka yang bersilangan:")
for _ in range(awal, akhir + 1):
    print(batasan_awal, batasan_akhir, end=' ')

    if angka1 > angka2:
        batasan_awal -= 1
        batasan_akhir += 1
    else:
        batasan_awal += 1
        batasan_akhir -= 1

    if awal != akhir:
        print("- ", end=' ')

print()

File: Praktikum 4. Loop/Soal 3/PRAK403-2310817310001-AvantioFierzaPatria.c

#include <stdio.h>

int main()
{
    int angka1, angka2;

    printf("Masukkan dua angka yang dipisahkan dengan spasi: ");
    scanf("%d %d", &angka1, &angka2);

    int batasan_awal, batasan_akhir, langkah, awal, akhir, i;
    char simbol = '-';

    batasan_awal = angka1;
    batasan_akhir = angka2;

    if (angka1 > angka2)
    {
        i = angka1 - angka2;
    }
    else
    {
        i = angka2 - angka1;
    }

    awal = 1;
    akhir = i + 1;
    printf("Angka yang bersilangan:\n");
    for (awal = 1; awal <= akhir; awal++)
    {
        printf("%d %d ", batasan_awal, batasan_akhir);

        if (angka1 > angka2)
        {
            batasan_awal -= 1;
            batasan_akhir += 1;
        }
        else
        {
            batasan_awal += 1;
            batasan_akhir -= 1;
        }
        if (awal != akhir)
        {
            printf("- ");
        }
    }
    printf("\n");
    return 0;
}
